packages/shared/src/components/modals/BootPopups.tsx

```diff
@@ -0,0 +1,178 @@
+  // hide modal if user already closed it
+  shouldHideStreaksModal =
+    shouldHideStreaksModal || alerts?.showStreakMilestone !== true;
+  // hide modal if there's no streak
+  shouldHideStreaksModal = shouldHideStreaksModal || !streak?.current;
+
+  const addBootPopup = (popup) => {
+    setBootPopups((prev) => new Map([...prev, [popup.type, popup]]));
+  };
+
+  /**
+   * Boot popup for reputation privileges.
+   */
+  useEffect(() => {
+    if (!isActionsFetched) {
+      return;
+    }
+
+    if (checkHasCompleted(ActionType.AckRep250)) {
+      return;
+    }
+
+    if (user?.reputation >= REP_TRESHOLD) {
+      addBootPopup({
+        type: LazyModal.ReputationPrivileges,
+        persistOnRouteChange: true,
+      });
+    }
+  }, [checkHasCompleted, isActionsFetched, openModal, user?.reputation]);
+
+  /**
+   * Boot popup based on marketing CTA
+   */
+  useEffect(() => {
+    if (marketingCta) {
+      addBootPopup({
+        type: LazyModal.MarketingCta,
+        props: {
+          marketingCta,
+          onAfterOpen: () => {
+            trackEvent({
+              event_name: AnalyticsEvent.Impression,
+              target_type: TargetType.MarketingCtaPopover,
+              target_id: marketingCta.campaignId,
+            });
+          },
+        },
+      });
+    }
+  }, [marketingCta, openModal, trackEvent]);
+
+  /** *
+   * Boot popup for generic referral campaign
+   */
+  useEffect(() => {
+    if (alerts?.showGenericReferral !== true) {
+      return;
+    }
+
+    addBootPopup({
+      type: LazyModal.GenericReferral,
+    });
+  }, [alerts?.showGenericReferral]);
+
+  /**
+   * Boot popup for migrate streaks
+   */
+  useEffect(() => {
+    if (!shouldShowStreaksPopup || !isActionsFetched) {
+      return;
+    }
+
+    addBootPopup({
+      type: LazyModal.MarketingCta,
+      props: {
+        marketingCta: promotion.migrateStreaks,
+        onAfterOpen: () => {
+          completeAction(ActionType.ExistingUserSeenStreaks);
+        },
+      },
+    });
+  }, [openModal, shouldShowStreaksPopup, isActionsFetched, completeAction]);
+
+  /**
+   * Boot popup for streaks milestone
+   */
+  useEffect(() => {
+    if (shouldHideStreaksModal) {
+      return;
+    }
+
+    openModal({
+      type: LazyModal.NewStreak,
+      props: {
+        currentStreak: streak?.current,
+        maxStreak: streak?.max,
+        onAfterClose: () => {
+          updateAlerts({ showStreakMilestone: false });
+        },
+      },
+    });
+  }, [
+    openModal,
+    shouldHideStreaksModal,
+    streak,
+    updateAlerts,
+    trackEvent,
+    user,
+  ]);
+
+  /**
+   * Actual rendering of the boot popup that's first in line
+   */
+  useEffect(() => {
+    if (
+      !loadedAlerts ||
+      !alerts?.bootPopup ||
+      bootPopups.size === 0 ||
+      shownModal
+    ) {
+      return;
+    }
+    openModal(bootPopups.values().next().value);
+    updateLastBootPopup();
+    setShownModal(true);
```

Comment:
Do you need this state, seems extra with all other conditions?

Reply:
Its more a extra for when no refresh is triggered to avoid showing a next one.
Felt more safe just in case, you reckon its overkill?

Reply:
More of an extra state. We can just do `setBootPopups(new Map())` to clear everything.